api/envoy/config/route/v3/route_components.proto

```diff
@@ -2398,5 +2397,7 @@ message FilterConfig {
   //   created and it is too late to change the chain.
```

Comment:
It isn't clear about when the disabled is true and is_optional is false, will envoy reject the config if the filter doesn't support.

Reply:
> If true, the filter is disabled in the route or virtual host and the `config` field is ignored.

I think the comment has clarified this scenario. The `config` will be ignored. So no any validation will be done and the config will not be rejected.